core/arch/arches.go

```diff
@@ -49,14 +48,70 @@ func (a Arches) String() string {
 	return strings.Join(a.set.SortedValues(), ",")
 }
 
-// ConstraintArch returns the arch for the constraint if there is one,
-// else it returns the default arch.
-func ConstraintArch(cons constraints.Value, defaultCons *constraints.Value) string {
-	if cons.HasArch() {
-		return *cons.Arch
+// The following constants define the machine architectures supported by Juju.
+const (
+	AMD64   = "amd64"
+	ARM64   = "arm64"
+	PPC64EL = "ppc64el"
+	S390X   = "s390x"
+	RISCV64 = "riscv64"
+)
+
+// AllSupportedArches records the machine architectures recognised by Juju.
+var AllSupportedArches = []string{
+	AMD64,
+	ARM64,
+	PPC64EL,
+	S390X,
+	RISCV64,
+}
+
+// UnsupportedArches records the machine architectures not supported by Juju.
+// Note: don't make const to prevent referencing it.
+var UnsupportedArches = []string{
+	"i386", "armhf",
```

Comment:
maybe include a few more in the list, the regexes for these include more variants.